dark/aa.py

```diff
@@ -1093,3 +1103,144 @@ def clustersForSequence(sequence, propertyNames, missingAAValue=0):
+
+    @param aaMatch: A C{dict} returned by C{compareAaReads}.
+    @param read1: A C{Read} instance or an instance of one of its subclasses.
+    @param read2: A C{Read} instance or an instance of one of its subclasses.
+    @param indent: A C{str} to indent all returned lines with.
+    @param offsets: If not C{None}, a C{set} of offsets of interest that were
+        only considered when making C{match}.
+    @return: A C{str} describing the match.
+    """
+    match = aaMatch['match']
+    MatchCount = match['MatchCount']
+    gapMismatchCount = match['gapMismatchCount']
+    gapGapMismatchCount = match['gapGapMismatchCount']
+    nonGapMismatchCount = match['nonGapMismatchCount']
+
+    if offsets:
+        len1 = len2 = len(offsets)
+    else:
+        len1, len2 = map(len, (read1, read2))
+
+    result = []
+    append = result.append
+
+    append(countPrint('%sMatches' % indent, MatchCount, len1, len2))
+    mismatchCount = (gapMismatchCount + gapGapMismatchCount +
+                     nonGapMismatchCount)
+    append(countPrint('%sMismatches' % indent, mismatchCount, len1, len2))
+    append(countPrint('%s  Not involving gaps (i.e., conflicts)' % (indent),
+                      nonGapMismatchCount, len1, len2))
+    append(countPrint('%s  Involving a gap in one sequence' % indent,
+                      gapMismatchCount, len1, len2))
+    append(countPrint('%s  Involving a gap in both sequences' % indent,
+                      gapGapMismatchCount, len1, len2))
+
+    for read, key in zip((read1, read2), ('read1', 'read2')):
+        append('%s  Id: %s' % (indent, read.id))
+        length = len(read)
+        append('%s    Length: %d' % (indent, length))
+        gapCount = len(aaMatch[key]['gapOffsets'])
+        append(countPrint('%s    Gaps' % indent, gapCount, length))
+        if gapCount:
+            append(
+                '%s    Gap locations (1-based): %s' %
+                (indent,
+                 ', '.join(map(lambda offset: str(offset + 1),
+                               sorted(aaMatch[key]['gapOffsets'])))))
+        extraCount = aaMatch[key]['extraCount']
+        if extraCount:
+            append(countPrint('%s    Extra nucleotides at end' % indent,
+                              extraCount, length))
+
+    return '\n'.join(result)
+
+
+def compareAaReads(read1, read2, gapChars='-',
+                   offsets=None):
+    """
+    Compare two amino acid sequences.
+
+    @param read1: A C{Read} instance or an instance of one of its subclasses.
+    @param read2: A C{Read} instance or an instance of one of its subclasses.
+    @param gapChars: An object supporting __contains__ with characters that
+        should be considered to be gaps.
+    @param offsets: If not C{None}, a C{set} of offsets of interest. Offsets
+        not in the set will not be considered.
+    @return: A C{dict} with information about the match and the individual
+        sequences (see below).
+    """
+    MatchCount = 0
```

Comment:
Don't use variables that start with an upper case letter. There's kind of a convention that you only use leading upper case for class names and for (variables that are intended to be) constants.

Reply:
Done